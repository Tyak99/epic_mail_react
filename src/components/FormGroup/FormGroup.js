import React from 'react';
import PropTypes from 'prop-types';
import style from './formGroup.css';

const FormGroup = ({
  type, placeholder, id, handleInput, name,
}) => (
  <div className={style.formGroup}>
    <input
      type={type}
      className={style.formControl}
      placeholder={placeholder}
      required
      id={id}
      onChange={handleInput}
      name={name}
    />
  </div>
);

FormGroup.defaultProps = {
  placeholder: '',
};
FormGroup.propTypes = {
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  id: PropTypes.string.isRequired,
  handleInput: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
};
export default FormGroup;
