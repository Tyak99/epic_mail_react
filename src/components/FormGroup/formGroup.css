.formGroup {
    margin-bottom: 40px;
  }
  .formControl {
    display: block;
    width: 100%;
    padding: 7px;
    font-size: 1em;
    font-weight: 400;
    border: none;
    border-bottom: 1px solid #ced4da;
    border-radius: 0.25em;
    background-color: #f1f1f1;
    outline: none;
  }
  