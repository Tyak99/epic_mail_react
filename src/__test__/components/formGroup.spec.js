import React from 'react';
import { shallow } from 'enzyme';
import FormGroup from '../../components/FormGroup/FormGroup';

const props = {
  type: 'text',
  id: 'inp',
  handleInput: () => {},
  name: 'text',
};
describe('component: FormGroup', () => {
  it('should render correctly', (done) => {
    const wrapper = shallow(<FormGroup {...props} />);
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
