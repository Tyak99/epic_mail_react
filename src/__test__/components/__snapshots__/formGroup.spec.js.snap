// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: FormGroup should render correctly 1`] = `ShallowWrapper {}`;
